soundcrate/src/app/api/lists/deleteSongList/route.js

```diff
@@ -0,0 +1,27 @@
+import { connectMongoDB } from '@/lib/mongodb';
+import SongList from '@/lib/models/songList';
+import { NextResponse } from 'next/server';
+
+export async function DELETE(req) {
+    try {
+        const { searchParams } = new URL(req.url);
+        const list_id = searchParams.get('id');
+
+        if (!list_id) {
+            return NextResponse.json({ message: "List ID is required" }, { status: 400 });
+        }
+
+        await connectMongoDB();
+
+        const deletedList = await SongList.findByIdAndDelete(list_id);
+
+        if (!deletedList) {
+            return NextResponse.json({ message: "List not found" }, { status: 404 });
+        }
+
```

Comment:
Let's add an additional check here to make sure that the list actually belongs to the currently logged in user, like return a 401 Unauthorized error instead. This can also be for another PR to add middleware to protect the deleteSongList and editSongList routes.

Reply:
i'll add the need to more protections to our slides instead of this merge for right now, since the buttons only show up if its the user's account.